sunpy/image/transform.py

```diff
@@ -304,6 +305,10 @@ def _rotation_skimage(image, matrix, shift, order, missing, clip):
         adjusted_image /= im_max
         adjusted_missing /= im_max
 
+    # Swap the byte order if it is non-native (e.g., big-endian on a little-endian system)
+    if adjusted_image.dtype.byteorder == ('>' if sys.byteorder == 'little' else '<'):
+        adjusted_image = adjusted_image.byteswap().newbyteorder()
```

Comment:
This does a full in mem copy right? That might want to be noted in the docstring?

Reply:
Note added